src/goto-programs/lazy_goto_model.h

```diff
@@ -243,6 +243,12 @@ class lazy_goto_modelt : public abstract_goto_modelt
     return goto_functions.at(id);
   }
 
+  goto_functionst::goto_functiont &get_goto_function_writable(
```

Comment:
:pick: Given your commit message:

> place entirely before symex it's ok if you're careful.

Probably worth doxygen comment explaining this danger (and if possible a check that symex hasn't begun??)